src/main/scala/ldbc/snb/datagen/spark/transform/GraphAssembler.scala

```diff
@@ -0,0 +1,158 @@
+        getByte(address, legacy.IP.BYTE2_SHIFT_POSITION),
+        getByte(address, legacy.IP.BYTE3_SHIFT_POSITION),
+        getByte(address, legacy.IP.BYTE4_SHIFT_POSITION)
+      )
+    }
+
+    val person = legacyPersons
+      .select(
+        temporalAttributes ++
+          Seq(
+            $"accountId".as("id"),
+            $"firstName",
+            $"lastName",
+            $"gender", // map to string
+            $"birthday", // map to date
+            formatIP($"ipAddress").as("locationIP"),
+            $"browserId".as("browserUsed") // join small dictionary
+          ): _*
+      )
+
+    val personEmailEmailAddress = legacyPersons
+      .select(
+        temporalAttributes ++
+          Seq(
+            $"accountId".as("Person.id"),
+            explode($"emails").as("email")
+          ): _*
+      )
+
+    val personSpeaksLanguage = legacyPersons
+      .select(
+        temporalAttributes ++
+          Seq(
+            $"accountId".as("Person.id"),
+            explode($"languages").as("language")
+          ): _*
+      )
+
+    val personHasInterestTag = legacyPersons
+      .select(
+        temporalAttributes ++ Seq(
+          $"accountId".as("Person.id"),
+          explode($"interests").as("Tag.id")
+        ): _*
+      )
+
+    val personIsLocatedInPlace = legacyPersons
+      .select(
+        temporalAttributes ++ Seq(
+          $"accountId".as("Person.id"),
+          $"cityId".as("Place.id")
+        ): _*
+      )
+
+    val personKnowsPerson = legacyPersons.
+      select(
+        Seq(
+          $"creationDate",
+          $"deletionDate",
+          $"isExplicitlyDeleted",
+          $"accountId",
+          explode($"knows").as("know")
+        ): _*
+      )
+      .select(
+        temporalAttributes ++ Seq(
+          $"accountId".as("Person1.id"),
+          $"know.to.accountId".as("Person2.id"),
+          $"know.weight".as("weight")
+        ): _*
+      )
+      .where($"`Person1.id`" < $"`Person2.id`")
+
+    //    val personLikesComment = ???
+    //
+    //    val personLikesPost = ???
+
+    val personStudyAtOrganisation = legacyPersons.
+      select(
+        temporalAttributes ++ Seq(
+          $"accountId".as("Person.id"),
+          $"universityLocationId".as("Organisation.id"), // join small dictionary
+          $"classYear" // format year
+        ): _*
+      )
+      .where($"`Organisation.id`".isNotNull) // -1 to null
+
+    val personWorkAtOrganisation = legacyPersons
+      .select(
+        temporalAttributes ++ Seq(
+          $"accountId".as("Person.id"),
+          explode($"companies").as(Seq("Organisation.id", "workFrom"))
+        ): _*
+      )
+
+    //    val forum = legacyActivities.
+    //      select(
+    //        explode($"_2.wall._1.creationDate").as($"creationDate")
+    //
+    //      )
+
+    Graph("Raw", Map(
+      Node("Person") -> person,
+      Attr("Email", "Person", "EmailAddress") -> personEmailEmailAddress,
```

Comment:
EmailEmail -> Email

Reply:
I gave this naming to have consistent naming across all the datasets, even if to sounds silly.